exonum-java-binding-core/rust/integration_tests/src/vm.rs

```diff
@@ -72,7 +73,16 @@ fn create_vm(debug: bool, with_fakes: bool) -> JavaVM {
         .build()
         .unwrap_or_else(|e| panic!("{:#?}", e));
 
-    JavaVM::new(jvm_args).unwrap_or_else(|e| panic!("{:#?}", e))
+    let vm = JavaVM::new(jvm_args).unwrap_or_else(|e| panic!("{:#?}", e));
+
+    // Initialize JNI cache for testing with fakes
+    if with_fakes {
+        // Current implementation of JavaVM#new() detaches current thread so we have to re-attach it
```

Comment:
Nit: this comment seems unnecessary given how the JNI API works in Rust (= Env guard)

Reply:
I put this comment because this part doesn't follow the general way of JNI API. I expected that thread that creates JVM should become registered (moreover - the main thread according to the specification). But I was a little surprised by the explicit detach just after JVM creation in the `jni_rs`:
```
            jni_error_code_to_result(sys::JNI_CreateJavaVM(
                &mut ptr as *mut _,
                &mut env as *mut *mut sys::JNIEnv as *mut *mut c_void,
                args.inner_ptr(),
            ))?;

            let vm = Self::from_raw(ptr)?;
            java_vm_unchecked!(vm.0, DetachCurrentThread);
```

Reply:
> that thread that creates JVM should become registered 

Dunno, when you register a thread explicitly, you get a guard. Here you get a VM and no guard.

Reply:
Ok, sounds reasonable...